src/sentry/dynamic_sampling/rules_generator.py

```diff
@@ -120,11 +103,18 @@ def generate_boost_release_rules(project_id: int, sample_rate: float) -> List[Re
                     "op": "and",
                     "inner": [
                         {
-                            "op": "glob",
+                            "op": "eq",
                             "name": "trace.release",
                             "value": [boosted_release.version],
                         },
-                        _generate_environment_condition(boosted_release.environment),
+                        {
+                            "op": "eq",
+                            "name": "trace.environment",
+                            # When environment is None, it will be mapped to equivalent null in json.
+                            # When Relay receives a rule with "value": [null] it will match it against events without
+                            # the environment tag set.
+                            "value": [boosted_release.environment],
```

Comment:
When the value in the event is `null`, `EqCondition` does not go into the `Array` branch, so I think we need just the value here: 
```suggestion
                            "value": boosted_release.environment,
```

https://github.com/getsentry/relay/blob/a1a12a06f1fdd53566965af4727dc6675268f5b5/relay-sampling/src/lib.rs#L66-L75

Reply:
good catch!

Reply:
This is why I wanted an ingest folk 👀, they know their baby well.